### PR TITLE
docs: Add pre-call filter example to cookbook

### PR DESCRIPTION
**Pull request by vikram-dagger**

### CODE REVIEW

File: docs/current_docs/cookbook/cookbook.mdx
Comment (vikram-dagger) [inline docs/current_docs/cookbook/cookbook.mdx:237]: @TomChv this does not seem to work. Can you help debug it?

Comment (TomChv) [reply docs/current_docs/cookbook/cookbook.mdx:237]: What's the directory content? Maybe you'll need to ignore `**/*` instead of `*`

Comment (TomChv) [reply docs/current_docs/cookbook/cookbook.mdx:237]: I wonder if it's because the source comes from a remote GH dir. It shouldn't matter though, can you also verify with a local dir to see if there's a difference?

File: docs/current_docs/cookbook/cookbook.mdx
Comment (TomChv) [inline docs/current_docs/cookbook/cookbook.mdx:202]: Did you write somewhere what's the difference between post & pre filtering? And which one is better depending on the case?

**Review by TomChv — COMMENTED** [review]
Content looks good, just left couple of questions

File: docs/current_docs/cookbook/cookbook.mdx
Comment (vikram-dagger) [reply docs/current_docs/cookbook/cookbook.mdx:237]: It works with a local dir, not with a remote dir. 

Repro:

```
admin@ip-172-31-41-139:~$ git clone https://github.com/dagger/dagger

admin@ip-172-31-41-139:~/sandbox/go$ dagger call copy-directory-with-exclusions --source=../../dagger directory --path=/src entries
✔ connect 0.2s
✔ initialize 1.0s
✔ prepare 0.1s
✔ myModule: MyModule! 0.0s
✔ MyModule.copyDirectoryWithExclusions(
    source: ✔ ModuleSource.resolveDirectoryFromCaller(ignore: ["*", "!**/*.md"], path: "../../dagger"): Directory! 0.0s
  ): Container! 1.2s
✔ Container.directory(path: "/src"): Directory! 0.0s
✔ Directory.entries: [String!]! 0.0s

.changes
.dagger
CHANGELOG.md
CODE_OF_CONDUCT.md
CONTRIBUTING.md
README.md
RELEASING.md
core
dagql
docs
hack
helm
modules
sdk

admin@ip-172-31-41-139:~/sandbox/go$ dagger call copy-directory-with-exclusions --source=https://github.com/dagger/dagger#main directory --path=/src entries
✔ connect 0.2s
✔ initialize 1.1s
✔ prepare 0.0s
✔ git(keepGitDir: true, url: "https://github.com/dagger/dagger"): GitRepository! 0.0s
✔ GitRepository.ref(name: "main"): GitRef! 0.0s
✔ GitRef.tree: Directory! 3.9s
✔ myModule: MyModule! 0.0s
✔ MyModule.copyDirectoryWithExclusions(
    source: ✔ GitRef.tree: Directory! 3.9s
  ): Container! 7.7s
✔ Container.directory(path: "/src"): Directory! 0.4s
✔ Directory.entries: [String!]! 0.0s

.changes
.changie.yaml
.dagger
.git
.github
.gitignore
.goreleaser.common.yml
.goreleaser.nightly.yml
.goreleaser.yml
.markdownlint.yaml
.readthedocs.yaml
.ruff.toml
CHANGELOG.md
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
NOTICE
README.md
RELEASING.md
analytics
auth
cmd
core
dagger.json
dagql
docs
engine
go.mod
go.sum
hack
helm
install.ps1
install.sh
internal
modules
network
sdk
testctx
version
yarn.lock
```

File: docs/current_docs/cookbook/cookbook.mdx
Comment (vikram-dagger) [reply docs/current_docs/cookbook/cookbook.mdx:202]: The cookbook is deliberately light on explanation, so I have added a link to the more detailed docs page discussing both pre- and post-call filters.

File: docs/current_docs/cookbook/cookbook.mdx
Comment (gerhard) [reply docs/current_docs/cookbook/cookbook.mdx]: I am seeing the same behaviour:

```console
# 👍 CONFIRMING DAGGER VERSION
❯ dagger version
dagger v0.13.7 (registry.dagger.io/engine:v0.13.7) darwin/amd64

# 👍 CONFIRMING PATH
❯ pwd
/Users/gerhard/github.com/dagger/dagger.worktree/8782/docs/current_docs/cookbook/snippets/copy-filter-dir-pre/go

# ✅ LOCAL DIR WORKS AS EXPECTED
❯ dagger call copy-directory-with-exclusions --source=../../../../../ directory --path=/src entries
✔ connect 0.3s
✔ initialize 0.8s
✔ prepare 0.1s
✔ myModule: MyModule! 0.0s
✔ MyModule.copyDirectoryWithExclusions(
    source: ✔ ModuleSource.resolveDirectoryFromCaller(ignore: ["*", "!**/*.md"], path: "../../../../../"): Directory! 0.0s
  ): Container! 0.9s
✔ Container.directory(path: "/src"): Directory! 0.0s
✔ Directory.entries: [String!]! 0.0s

ARCHIVING.md
STYLE_GUIDE.md
_README.md
docs-graphql

# ❌ REMOTE DIR via GIT DOES NOT WORK AS EXPECTED
❯ dagger call copy-directory-with-exclusions --source="https://github.com/dagger/dagger#main" directory --path=/src entries
✔ connect 0.3s
✔ initialize 0.8s
✔ prepare 0.0s
✔ host: Host! 0.0s
✔ Host.unixSocket(path: "/Users/gerhard/.1password/agent.sock"): Socket! 0.0s
✔ git(
    keepGitDir: true
    sshAuthSocket: ✔ Host.unixSocket(path: "/Users/gerhard/.1password/agent.sock"): Socket! 0.0s
    url: "https://github.com/dagger/dagger"
  ): GitRepository! 0.0s
✔ GitRepository.ref(name: "main"): GitRef! 0.0s
✔ GitRef.tree: Directory! 0.3s
✔ myModule: MyModule! 0.0s
✔ MyModule.copyDirectoryWithExclusions(
    source: ✔ GitRef.tree: Directory! 0.3s
  ): Container! 1.2s
✔ Container.directory(path: "/src"): Directory! 0.0s
✔ Directory.entries: [String!]! 0.0s

.changes
.changie.yaml
.dagger
.git
.github
.gitignore
.goreleaser.common.yml
.goreleaser.nightly.yml
.goreleaser.yml
.markdownlint.yaml
.readthedocs.yaml
.ruff.toml
CHANGELOG.md
CODE_OF_CONDUCT.md
CONTRIBUTING.md
LICENSE
NOTICE
README.md
RELEASING.md
analytics
auth
cmd
core
dagger.json
dagql
docs
engine
go.mod
go.sum
hack
helm
install.ps1
install.sh
internal
modules
network
sdk
testctx
version
yarn.lock
```

I pushed my local snippet change @vikram-dagger , but feel free to drop it if not needed.

Comment (vikram-dagger) [reply docs/current_docs/cookbook/cookbook.mdx]: Thanks for re-checking this @gerhard !

Comment (TomChv) [reply docs/current_docs/cookbook/cookbook.mdx]: I also did a repro, seems like an established issue, I'm going to work on a fix!
We can hold this PR until the fixes is merged

**Review by TomChv — APPROVED** [review]

**Pull request merged**
